x-pack/plugins/reporting/server/plugin.ts

```diff
@@ -68,20 +62,21 @@ export class ReportingPlugin
     // use data plugin for csv formats
     setFieldFormats(plugins.data.fieldFormats);
 
-    const { logger } = this;
-    const reportingCore = this.getReportingCore();
-    const config = reportingCore.getConfig();
+    const { logger, reportingCore } = this;
     const { elasticsearch } = reportingCore.getPluginSetupDeps();
 
     // async background start
     (async () => {
+      await this.reportingCore.pluginIsSetup();
```

Comment:
This is to ensure that by the time we get past this line, the config is available.